src/topology/bornology/basic.lean

```diff
@@ -0,0 +1,152 @@
+/-
+Copyright (c) 2022 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import order.filter.cofinite
+
+/-!
+# Basic theory of bornology
+
+We develop the basic theory of bornologies. Instead of axiomatizing bounded sets and defining
+bornologies in terms of those, we recognize that the cobounded sets form a filter and define a
+bornology as a filter of cobounded sets which contains the cofinite filter.  This allows us to make
+use of the extensive library for filters, but we also provide the relevant connecting results for
+bounded sets.
+
+The specification of a bornology in terms of the cobounded filter is equivalent to the standard
+one (e.g., see [Bourbaki, *Topological Vector Spaces*][bourbaki1987], **covering bornology**, now
+often called simply **bornology**) in terms of bounded sets (see `bornology.of_bounded`,
+`is_bounded.union`, `is_bounded.subset`), except that we do not allow the empty bornology (that is,
+we require that *some* set must be bounded; equivalently, `∅` is bounded).
```

Comment:
Since you're bringing this up, could you mention that the cobounded filter is known in the literature as the "filter at infinity"? (If I've recalled the terminology correctly.)